internal/consensus/state.go

```diff
@@ -2254,7 +2254,54 @@ func (cs *State) tryAddVote(vote *types.Vote, peerID p2p.ID) (bool, error) {
 	return added, nil
 }
 
+// addVote adds a vote to the consensus state. It performs various checks and updates the state accordingly.
 func (cs *State) addVote(vote *types.Vote, peerID p2p.ID) (added bool, err error) {
+	cs.logVoteDetails(vote)
+
+	if cs.isLateVote(vote) {
+		cs.metrics.MarkLateVote(vote.Type)
+	}
+
+	// A precommit for the previous height?
+	// These come in while we wait timeoutCommit
+	if cs.isPrecommitForPreviousHeight(vote) {
+		return cs.handlePrecommitForPreviousHeight(vote)
+	}
+
+	// Height mismatch is ignored.
+	// Not necessarily a bad peer, but not favorable behavior.
+	if cs.isHeightMismatch(vote) {
```

Comment:
How is this code easier to read than `vote.Height != cs.Height`?

Reply:
english.  This version calls a method whose name is self descriptive.  

But to change it back is nbd